File: Programmers/Lv0/배열 만들기4.py
# 내 풀이
def solution(arr):
    stk = []
    i=0
    while i<len(arr):
        if len(stk)==0:
            stk.append(arr[i])
            i+=1
        else:
            if stk[-1]<arr[i]:
                stk.append(arr[i])
                i+=1
            else:
                stk.pop()
    return stk

# 다른 사람 풀이
    def solution(arr):
        stk = []
        for i in range(len(arr)):
            while stk and stk[-1] >= arr[i]:
                stk.pop()
            stk.append(arr[i])
        return stk

File: Programmer_Java/Lv0/배열 만들기4.java
// package Programmers;

// public class 배열 만들기4 {
  
// }

//출처:프로그래머스,
//https://school.programmers.co.kr/learn/courses/30/lessons/181918

/*
 * // list 정의
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> result = new ArrayList<>();
        int i = 0;
        while(i<arr.length){
            if(result.size()==0){
                result.add(arr[i]);
                i++;
            }
            else{
                if( result.get(result.size()-1) <arr[i]){
                    result.add(arr[i]);
                    i++;
                }
                else{
                    result.remove(result.size()-1);
                }
            }
            // System.out.println(result);
        }
        // System.out.print(result);
        int[] stk = new int[result.size()];
        for(int k =0; k<result.size(); k++){
            stk[k]=result.get(k);
        }
        
        return stk;
    }
}
 */